### PR TITLE
Stac 21209 export relations from servicegraph metrics

### PR DESCRIPTION
**Pull request by fvlankvelt**

### CODE REVIEW

File: exporter/ststopologyexporter/exporter_test.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ func TestExporter_pushResourcesData(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, 1, len(payload.Topologies))
 		require.Equal(t, 3, len(payload.Topologies[0].Components))
+		require.Equal(t, 2, len(payload.Topologies[0].Relations))
```
Comment (rb3ckers) [inline exporter/ststopologyexporter/exporter_test.go:27]: Would also be good to check, for both component and relation, that the expected attributes are present and others are not (like no API key but there is a service.name)

File: exporter/ststopologyexporter/internal/topology_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			SourceId:   "urn:opentelemetry:namespace/demo:service/demo 1",
+			TargetId:   "urn:opentelemetry:namespace/demo:service/demo 1:serviceInstance/demo 1",
+			Type: RelationType{
+				Name: "provided by",
+			},
+			Data: &RelationData{
+				Tags: map[string]string{},
+			},
+		},
+	}, relations)
```
Comment (rb3ckers) [inline exporter/ststopologyexporter/internal/topology_test.go:86]: I would also add a test for `AddConnection`. There is enough logic in there that warrants a test, even if it is just against typos and to capture the intent of the code.

File: exporter/ststopologyexporter/internal/topology.go [hunk 3]
```diff
@@ -43,40 +45,95 @@ func (c *ComponentsCollection) AddResource(attrs *pcommon.Map) bool {
...
+
+	instanceId, ok = attrs.Get("server_service.instance.id")
+	var serverInstanceId string
+	if !ok {
+		serverInstanceId = reqAttrs["server"]
+	} else {
+		serverInstanceId = instanceId.AsString()
+	}
+	targetId := fmt.Sprintf("urn:opentelemetry:namespace/%s:service/%s:serviceInstance/%s", reqAttrs["server_service.namespace"], reqAttrs["server"], serverInstanceId)
+	c.addRelation(sourceId, targetId, reqAttrs["connection_type"])
```
Comment (rb3ckers) [inline exporter/ststopologyexporter/internal/topology.go:120]: I'm pretty sure the `connection_type` is an empty string for anything that's a "normal" connection and not a database, messaging_system or virtual_node.

File: exporter/ststopologyexporter/exporter_test.go (hunk 1 above)
Comment (fvlankvelt) [reply exporter/ststopologyexporter/exporter_test.go:38]: wanted to do only very basic checks here and do the detailed verification in the "topology test".  Makes sense to check for the `sts_api_key`, so I've added that.

File: exporter/ststopologyexporter/internal/topology.go (hunk 3 above)
Comment (fvlankvelt) [reply exporter/ststopologyexporter/internal/topology.go]: ah, indeed.

**Review by rb3ckers — APPROVED** [review]

**Pull request merged**